src/sentry/search/events/fields.py

```diff
@@ -2265,13 +2275,63 @@ def __init__(self, dataset: Dataset, params: ParamsType):
                     ),
                     default_result_type="percentage",
                 ),
+                SnQLFunction(
+                    "percentile",
+                    required_args=[
+                        SnQLColumnOrAlias("column"),
+                        NumberRange("percentile", 0, 1),
+                    ],
+                    snql_aggregate=self._resolve_percentile,
+                    result_type_fn=reflective_result_type(),
+                    default_result_type="duration",
+                    redundant_grouping=True,
+                ),
+                SnQLFunction(
+                    "p50",
+                    optional_args=[
+                        with_default("transaction.duration", SnQLColumnOrAlias("column")),
+                    ],
+                    snql_aggregate=lambda args, alias: self._resolve_percentile(args, alias, 0.5),
+                    default_result_type="integer",
+                    redundant_grouping=True,
+                ),
+                SnQLFunction(
+                    "p75",
+                    optional_args=[
+                        with_default("transaction.duration", SnQLColumnOrAlias("column")),
+                    ],
+                    snql_aggregate=lambda args, alias: self._resolve_percentile(args, alias, 0.75),
+                    default_result_type="integer",
+                    redundant_grouping=True,
+                ),
+                SnQLFunction(
+                    "p95",
+                    optional_args=[
+                        with_default("transaction.duration", SnQLColumnOrAlias("column")),
+                    ],
+                    snql_aggregate=lambda args, alias: self._resolve_percentile(args, alias, 0.95),
+                    default_result_type="integer",
+                    redundant_grouping=True,
+                ),
+                SnQLFunction(
+                    "p99",
+                    optional_args=[
+                        with_default("transaction.duration", SnQLColumnOrAlias("column")),
+                    ],
+                    snql_aggregate=lambda args, alias: self._resolve_percentile(args, alias, 0.99),
+                    default_result_type="integer",
+                    redundant_grouping=True,
+                ),
+                SnQLFunction(
+                    "p100",
+                    optional_args=[
+                        with_default("transaction.duration", SnQLColumnOrAlias("column")),
+                    ],
+                    snql_aggregate=lambda args, alias: self._resolve_percentile(args, alias, 1),
+                    default_result_type="integer",
+                    redundant_grouping=True,
```

Comment:
All these functions should support `reflexive_result_type()` eventually, not just `percentile`.

Reply:
Updated to `reflective_result_type()`!